src/util/mpir_pmi.c

```diff
@@ -549,7 +549,7 @@ static int put_ex(const char *key, const void *buf, int bufsize, int is_local)
         MPIR_ERR_CHECK(mpi_errno);
         for (int i = 0; i < num_segs; i++) {
             char seg_key[50];
-            sprintf(seg_key, "%s-seg-%d/%d", key, i + 1, num_segs);
```

Comment:
NAK: this is a backwards incompatible change

Reply:
We can just allow `/` in the key. I'll make the change